assets/js/frontend/single-product.js

```diff
@@ -154,68 +174,81 @@ jQuery( function( $ ) {
 			// But only zoom if the img is larger than its container.
 			if ( enable_zoom ) {
 				zoom_target.trigger( 'zoom.destroy' );
-				zoom_target.zoom({
+				zoom_target.zoom( {
 					touch: false
-				});
+				} );
 			}
-		},
+		};
+
+		this.reset_slide_position = function() {
+
+			if ( ! gallery.flexslider_enabled ) {
+				return;
+			}
+
+			$el.flexslider( 0 );
+		};
 
 		/**
 		 * Get product gallery image items.
 		 */
-		get_gallery_items: function() {
-			var $slides = $( '.woocommerce-product-gallery__wrapper' ).children(),
-				items   = [],
-				index   = $slides.filter( '.' + 'flex-active-slide' ).index();
+		this.get_gallery_items = function() {
+
+			var $slides = this.$images,
+				items   = [];
 
 			if ( $slides.length > 0 ) {
 				$slides.each( function( i, el ) {
 					var img = $( el ).find( 'img' ),
-						large_image_src = img.attr( 'data-large-image' ),
-						large_image_w   = img.attr( 'data-large-image-width' ),
-						large_image_h   = img.attr( 'data-large-image-height' ),
+						large_image_src = img.attr( 'data-large_image' ),
+						large_image_w   = img.attr( 'data-large_image_width' ),
+						large_image_h   = img.attr( 'data-large_image_height' ),
 						item            = {
 							src: large_image_src,
 							w:   large_image_w,
 							h:   large_image_h,
 							title: img.attr( 'title' )
 						};
 					items.push( item );
-				});
+				} );
 			}
 
-			return {
-				index: index,
-				items: items
-			};
-		},
+			return items;
```

Comment:
The `index` prop of the returned object is not used anywhere, so why return an object here? Also since we want to be able to enable specific features, the `index` variable should not be discovered using `flex-active-slide`, which assumes that the flexslider feature is active.

Now this simply returns the gallery items.